src/server/saved_objects/client/lib/create_find_query.js

```diff
@@ -35,5 +40,20 @@ export function createFindQuery(options = {}) {
     });
   }
 
-  return { version: true, query: { bool } };
+  const query = { version: true, query: { bool } };
+
+  if (sortField) {
+    const value = {
+      order: sortOrder,
+      unmapped_type: get(mappings, `${type}.properties.${sortField}.type`)
```

Comment:
For v6 mappings, this path would be `doc.properties.${type}.properties.${sortField}.type`

Could look something like this:

```
    const v5MappingType = get(mappings, `${type}.properties.${sortField}.type`);
    const v6MappingType = get(mappings, `doc.properties.${type}.properties.${sortField}.type`);
    const mappingType = v5MappingType || v6MappingType;

    const value = {
      order: sortOrder,
      unmapped_type: mappingType
    };
```